src/NKSolver/NKSolvers.F90

```diff
@@ -3166,13 +3220,24 @@ subroutine physicalityCheckANK(lambdaP)
     call EChk(ierr,__FILE__,__LINE__)
 
     ! Make sure that we did not get any NaN's in the process
-    if (isnan(lambdaL)) lambdaL = zero
+    if (isnan(real(lambdaL))) then
```

Comment:
good catch. I think this is left like this because I had `lambdaL` as a `realType` previously. Now that its `alwaysRealType`, we dont need this check. Removing it